### DISCUSSION TITLE
Compile lotus from source

### DISCUSSION THREAD

**Discussion by de-authority** [filecoin-project/lotus#7354] [Tutorials]
We already have doc about how to compile lotus. you should follow the **[docs](https://docs.filecoin.io/get-started/lotus/installation/#building-from-source)**  
Though. i want add some tips here

### 1. You should chose one of those bellow options. you don't have to compile every option. 
- compile for mainnet, run:     
```sh
make clean all 
```  

- compile for devnet, run:    
```sh
make clean 2k
```  

- compile for calibration net,run:    
```sh
make clean calibnet
```

### 2. Use vpn if you are in China
You should use vpn if you want a good experience when compile. and replace rust repository if needed. you can see doc [here](https://blog.csdn.net/xiangxianghehe/article/details/105874880)

### 3. this is not error when you run 'sudo make install'
>bash: go: command not found  
expr: syntax error: unexpected argument ‘1016000’  
install -C ./lotus /usr/local/bin/lotus  
install -C ./lotus-miner /usr/local/bin/lotus-miner  
install -C ./lotus-worker /usr/local/bin/lotus-worker  

why "bash: go: command not found"?   
Typically  go is installed at /usr/local/go/bin/go. but system will not search /usr/local/go/bin/ when you use sudo.

why "expr: syntax error: unexpected argument ‘1016000’"  
Same reason as prior one. caused by this [**line**](https://github.com/filecoin-project/lotus/blob/master/Makefile#L11) of Makefile

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
there is no 2k network

**Reply by de-authority** [reply]
It's devnet.  Is this description misleading and what do you think needs to be changed？
